ouroboros-network/src/Ouroboros/Network/PeerSelection/RootPeersDNS.hs

```diff
@@ -444,19 +447,17 @@ localRootPeersProvider tracer
             Right results -> do
               atomically $ do
                 rootPeersGroups <- readTVar rootPeersGroupsVar
-                let rootPeersGroups' = IntMap.fromList $ zip [0,1..] rootPeersGroups
-                    (target, entry)  = rootPeersGroups' IntMap.! group
+                let (target, entry)  = rootPeersGroups `Seq.index` index
                     resultsMap       = Map.fromList (map fst results)
                     entry'           = entry <> resultsMap
```

Comment:
^ shouldn't this be `resultsMap <> entry`, `Map.union` is left biased?

Reply:
Ah! I think you're right, good catch! Thank you!